pkg/util/archive_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+)
+
+func writeFileOrDie(path string, data []byte, t *testing.T) {
+	if err := ioutil.WriteFile(path, data, os.FileMode(0644)); err != nil {
+		t.Errorf("unexpected error: %v", err)
+		t.FailNow()
+		return
+	}
+
+}
+
+func TestArchive(t *testing.T) {
+	expected := map[string]string{
+		"file1": "foobar",
+		"file2": "baz",
+		"file3": "blah",
+	}
+
+	tmp, err := ioutil.TempDir(os.TempDir(), "archive")
+	defer os.RemoveAll(tmp)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+		t.FailNow()
+		return
+	}
+	if err := os.MkdirAll(tmp, os.FileMode(0755)); err != nil {
+		t.Errorf("unexpected error: %v", err)
+		t.FailNow()
+		return
+	}
+	for file, data := range expected {
+		writeFileOrDie(path.Join(tmp, file), []byte(data), t)
+	}
+
+	for _, compress := range []bool{true, false} {
```

Comment:
This would be easier to read as a set of table driven tests:

```
tests := []struct {
  compress, stream bool
}{
  {true, true},
  {true, false},
  {false, true},
  {false, false},
}
```